### ISSUE TITLE
how to  calculate absolute coordinates

### ISSUE THREAD

**Issue by bin7777** [MCG-NJU/MixFormerV2#4]
`        coord_l = torch.sum((self.indice * prob_l), dim=-1)
        coord_r = torch.sum((self.indice * prob_r), dim=-1)
        coord_t = torch.sum((self.indice * prob_t), dim=-1)
        coord_b = torch.sum((self.indice * prob_b), dim=-1) `
Is the 'prob' distribution probability of coordinates?why calculate the sum

**Comment by songtianhui** [comment]
Yes,`prob` is the probability distribution of coordinates. The weighted sum is expectation over the coordinates which is result.

**Comment by songtianhui** [comment]
The problem you mentioned actually does exist, the model learn to look for the boundary of target. Even I have seen this phenomenon occasionally when there are similar objects in the search region, the probability heat map of boundary will show double or more peaks. But it works well in most instances to locate uniquely the target.
Btw, it can be the point that can further optimize the algorithm.

**Comment by bin7777** [comment]
thanks
